Uploaded homework 7

ErnestMihasenko opened the pull request:
Got some problems with creating a variable with the value of class

ErnestMihasenko committed:
Uploaded homework 7

Got some problems with creating a variable with the value of class

darya-gurinovich commented on homework7.playground/Contents.swift:
```diff
@@ -0,0 +1,44 @@
+    }
+    func printInfo() {
+        let info = students
+        print(info)
```
You need to print all students info. This way you just print the array type (I believe it would print something like "Array<Student>"). You need to call your implemented `printInfo` method for every student here

darya-gurinovich commented on homework7.playground/Contents.swift:
```diff
@@ -0,0 +1,44 @@
+    
+    init(name: String,
+         surname: String,
+         marks: [String: Int] = ["Нет оценок": 0],
```
You shoudn't have "Нет оценок" in the dictionary. You just need an empty dictionary to cover this case. So it's better to replace this with ` marks: [String: Int] = [:]`

darya-gurinovich commented on homework7.playground/Contents.swift:
```diff
@@ -0,0 +1,44 @@
+    init(name: String,
+         surname: String,
+         marks: [String: Int] = ["Нет оценок": 0],
+         grade: Int = 0) {
```
I believe the smallest grade that we have is 1

darya-gurinovich commented on homework7.playground/Contents.swift:
```diff
@@ -0,0 +1,44 @@
+    }
+}
+
+// let school = School()
```
You need to pass `name` and `surname` because they're set in your init and don't have a default value like `marks` and `grade`

darya-gurinovich reviewed (changes requested):
Good effort! Good code style and naming. Please, resolve my comments and we're good to go

ErnestMihasenko committed:
Fixed previous issues

Added default values to all properties of class "Student", but still can't create a varible with School class value. So, I suppose, there is another problem. Maybe it is related to playground somehow... If you have another ideas, pls, tell me)

darya-gurinovich commented on homework7.playground/Contents.swift:
```diff
@@ -0,0 +1,45 @@
+        self.students = students
+    }
+    func printInfo() {
+        let info = [firstStudent, secondStudent, thirdStudent, fourthStudent]
+        print(info)
```
You should use only `students` property inside the class. This function will stop working if you pass other data to `students`. You need to go through your `students` with cycle and call `printInfo()` in every iteration

darya-gurinovich commented on homework7.playground/Contents.swift:
```diff
@@ -0,0 +1,45 @@
+let firstStudent = Student(name: "Павлик", surname: "Волков", marks: ["Математика": 3], grade: 5)
+let secondStudent = Student(name: "Маша", surname: "Волкова", marks: ["Математика": 3], grade: 5)
+let thirdStudent = Student(name: "Антон", surname: "Липень", marks: ["Английский": 5, "Русский": 5, "Физика": 5], grade: 5)
+let fourthStudent = Student(name: "Алена", surname: "Воронина", marks: ["Английский": 9, "Русский": 7, "Физика": 4], grade: 8)
```
I can't see where you create `School` object. You need to create it and call `printInfo()` for it

ErnestMihasenko committed:
Decided to do it in project

Now it works)

ErnestMihasenko committed:
Decided to do it in project

Now it works)

ErnestMihasenko committed:
Delete Lesson 7 homework.xcodeproj directory

darya-gurinovich commented on Lesson 7 homework/Lesson 7 homework/ViewController.swift:
```diff
@@ -0,0 +1,62 @@
+    }
+    func printInfo() {
+        for info in [firstStudent.printInfo(), secondStudent.printInfo(), thirdStudent.printInfo(), fourthStudent.printInfo()] {
+        print(info)
```
Please, add a tab here

darya-gurinovich commented on Lesson 7 homework/Lesson 7 homework/ViewController.swift:
```diff
@@ -0,0 +1,62 @@
+        self.students = students
+    }
+    func printInfo() {
+        for info in [firstStudent.printInfo(), secondStudent.printInfo(), thirdStudent.printInfo(), fourthStudent.printInfo()] {
+        print(info)
+        }
```
You CAN'T access `firstStudent`, `secondStudent`, `thirdStudent`, `fourthStudent`. You should ONLY work with `students` property. 

You don't need to create a new array and definitely don't create an array with `printInfo()`. You need to call it inside the cycle on your `info` property

darya-gurinovich reviewed (changes requested):
Please, resolve your `printInfo` in School issues. If you have questions, please, feel free to contact me through our group chat

ErnestMihasenko committed:
Fixed the last comments

darya-gurinovich commented on Lesson 7 homework/Lesson 7 homework/ViewController.swift:
```diff
@@ -0,0 +1,61 @@
+        self.surname = surname
+        self.marks = marks
+        self.grade = grade
+        }
```
This `}` should be one tab to the left

darya-gurinovich replied:
Please, add an empty line between functions

darya-gurinovich commented on Lesson 7 homework/Lesson 7 homework/ViewController.swift:
```diff
@@ -0,0 +1,61 @@
+        school.printInfo()
+    }
+}
+
+import UIKit
+import Foundation
```
I believe you can remove these imports

darya-gurinovich commented on Lesson 7 homework/Lesson 7 homework/ViewController.swift:
```diff
@@ -0,0 +1,61 @@
+    }
+}
+
+let school = School()
```
It's better to move this variable creation to `viewDidLoad()`

darya-gurinovich commented on Lesson 7 homework/Lesson 7 homework/ViewController.swift:
```diff
@@ -0,0 +1,61 @@
+    }
+}
+
+let firstStudent = Student(name: "Павлик", surname: "Волков", marks: ["Математика": 3], grade: 5)
+let secondStudent = Student(name: "Маша", surname: "Волкова", marks: ["Математика": 3], grade: 5)
+let thirdStudent = Student(name: "Антон", surname: "Липень", marks: ["Английский": 5, "Русский": 5, "Физика": 5], grade: 5)
+let fourthStudent = Student(name: "Алена", surname: "Воронина", marks: ["Английский": 9, "Русский": 7, "Физика": 4], grade: 8)
```
It's better to move these variables creation to `viewDidLoad()` as well

darya-gurinovich reviewed (approved):
Good work! Well done! Please, remove your playground files and leave only xproj files

ErnestMihasenko committed:
Delete homework7.playground directory
